Build fails with "lib/resolver.c:14:10: fatal error: 'ares.h' file not found" in 8.0

**Describe the bug**

The build fails in `vtysh/extract.pl` when libcares is not installed due to missing `ares.h` for `lib/resolver.c`.

[x] Did you check if this is a duplicate issue?
[x] Did you test it on the latest FRRouting/frr master branch?

**To Reproduce**

1. Ensure c-ares is not installed
2. Run configure, disabling nhrpd nor bmp to avoid triggering explicit dependency
3. Run make

**Expected behavior**

Build succeeds. c-ares dependency is truly optional.

**Screenshots**

```
vtysh/extract.pl lib/distribute.c lib/filter.c lib/filter_cli.c lib/if.c lib/if_rmap.c lib/keychain.c lib/lib_vty.c lib/log_vty.c lib/nexthop_group.c lib/plist.c lib/resolver.c lib/routemap.c lib/routemap_cli.c lib/spf_backoff.c lib/thread.c lib/vrf.c lib/vty.c lib/agentx.c zebra/debug.c zebra/interface.c zebra/router-id.c zebra/rtadv.c zebra/zebra_gr.c zebra/zebra_mlag_vty.c zebra/zebra_evpn_mh.c zebra/zebra_mpls_vty.c zebra/zebra_ptm.c zebra/zebra_pw.c zebra/zebra_routemap.c zebra/zebra_vty.c zebra/zserv.c zebra/irdp_interface.c zebra/zebra_fpm.c watchfrr/watchfrr_vty.c bgpd/bgp_bfd.c bgpd/bgp_debug.c bgpd/bgp_dump.c bgpd/bgp_evpn_mh.c bgpd/bgp_evpn_vty.c bgpd/bgp_filter.c bgpd/bgp_labelpool.c bgpd/bgp_mplsvpn.c bgpd/bgp_nexthop.c bgpd/bgp_route.c bgpd/bgp_routemap.c bgpd/bgp_vty.c bgpd/bgp_flowspec_vty.c bgpd/bgp_rpki.c bgpd/bgp_bmp.c bgpd/rfapi/bgp_rfapi_cfg.c bgpd/rfapi/rfapi.c bgpd/rfapi/rfapi_vty.c bgpd/rfapi/vnc_debug.c ripd/rip_cli.c ripd/rip_debug.c ripd/ripd.c ripngd/ripng_cli.c ripngd/ripng_debug.c ripngd/ripngd.c ospfd/ospf_bfd.c ospfd/ospf_dump.c ospfd/ospf_ldp_sync.c ospfd/ospf_opaque.c ospfd/ospf_ri.c ospfd/ospf_routemap.c ospfd/ospf_te.c ospfd/ospf_sr.c ospfd/ospf_vty.c ospf6d/ospf6_abr.c ospf6d/ospf6_asbr.c ospf6d/ospf6_area.c ospf6d/ospf6_bfd.c ospf6d/ospf6_flood.c ospf6d/ospf6_interface.c ospf6d/ospf6_intra.c ospf6d/ospf6_lsa.c ospf6d/ospf6_message.c ospf6d/ospf6_neighbor.c ospf6d/ospf6_route.c ospf6d/ospf6_spf.c ospf6d/ospf6_top.c ospf6d/ospf6_zebra.c ospf6d/ospf6d.c isisd/isis_cli.c isisd/isis_ldp_sync.c isisd/isis_redist.c isisd/isis_spf.c isisd/isis_te.c isisd/isis_sr.c isisd/isis_vty_fabricd.c isisd/isisd.c babeld/babel_interface.c babeld/babel_zebra.c babeld/babeld.c eigrpd/eigrp_cli.c eigrpd/eigrp_dump.c eigrpd/eigrp_vty.c pbrd/pbr_vty.c pbrd/pbr_debug.c staticd/static_vty.c bfdd/bfdd_vty.c bfdd/bfdd_cli.c pathd/path_cli.c pathd/path_ted.c pathd/path_pcep_cli.c --have-isisd --have-fabricd > vtysh/vtysh_cmd.c
lib/resolver.c:14:10: fatal error: 'ares.h' file not found
#include <ares.h>
         ^~~~~~~~
1 error generated.
```

**Versions**
 - OS Version: FreeBSD 12.2
 - Kernel: FreeBSD 12.2
 - FRR Version: 8.0

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/lib/subdir.am
+++ b/lib/subdir.am
@@ -144,7 +144,6 @@ vtysh_scan += \
 	lib/log_vty.c \
 	lib/nexthop_group.c \
 	lib/plist.c \
-	lib/resolver.c \
 	lib/routemap.c \
 	lib/routemap_cli.c \
 	lib/spf_backoff.c \
@@ -335,6 +334,7 @@ lib_libfrrsnmp_la_SOURCES = \
 if CARES
 lib_LTLIBRARIES += lib/libfrrcares.la
 pkginclude_HEADERS += lib/resolver.h
+vtysh_scan += lib/resolver.c
 endif
 
 lib_libfrrcares_la_CFLAGS = $(AM_CFLAGS) $(CARES_CFLAGS)

